Sources/SentryCrash/Recording/Tools/SentryHook.c

```diff
@@ -56,6 +28,80 @@ sentrycrash__async_backtrace_decref(sentrycrash_async_backtrace_t *bt)
     }
 }
 
+/**
+ * This is a poor-mans concurrent hashtable.
+ * We have N slots, using modulo of the thread ID. Using atomic load / compare-exchange to make sure
+ * that the slot indeed belongs to the thread we want to work with.
+ */
+
+#define SENTRY_MAX_ASYNC_THREADS (128 - 1)
+
+typedef struct {
+    SentryCrashThread thread;
+    sentrycrash_async_backtrace_t *backtrace;
+} sentrycrash_async_caller_t;
+
+static sentrycrash_async_caller_t sentry_async_callers[SENTRY_MAX_ASYNC_THREADS] = { 0 };
```

Comment:
I assume you're keen to avoid a global mutex on this table?  Would a mutex on each bucket be reasonable?  It would make reasoning about what is going on in a bucket a lot easier I think.

Also, if you did you could potentially put a list of threads in each bucket instead of just one so you don't have such a hard limit on thread count and you're not affected by hash collisions

Reply:
This code is perf sensitive as it is, so I wouldn’t want to have either a global, or a per-bucket mutex.